manager/orchestrator/task.go

```diff
@@ -67,7 +67,29 @@ func IsTaskDirty(s *api.Service, t *api.Task) bool {
 		return false
 	}
 
-	return !reflect.DeepEqual(s.Spec.Task, t.Spec) ||
+	// Make a deep copy of the service and task spec for the comparison.
+	serviceTaskSpec := *s.Spec.Task.Copy()
```

Comment:
We discussed this offline but posting here to follow-up: The call to `DeepEqual()` later on is comparing it to a non-pointer `api.TaskSpec`. I could have either dereferenced it here or there, but what's important is that it is still calling `DeepEqual()` with the same types.